libp2p/protocols/pubsub/floodsub.nim

```diff
@@ -96,7 +96,13 @@ method rpcHandler*(f: FloodSub,
     f.handleSubscribe(peer, sub.topic, sub.subscribe)
 
   for msg in rpcMsg.messages:                         # for every message
-    let msgId = f.msgIdProvider(msg)
+    let msgIdResult = f.msgIdProvider(msg)
+    if msgIdResult.isErr:
+      debug "Dropping message due to failed message id generation",
```

Comment:
although it doesn't necessarily have to be part of this PR, we should at least note in a TODO that we need to descore peers for which reject is returned here.

Also, seeing this code makes me think of another potential polishing feature for the future: turn `ValidationResult` into `ValidationError` without accept, because `err(ValidationResult.Accept` is not really meaningful.

regarding backwards compat, that can actually be done quite easily @Menduist - the enrolling function that takes a validation callback can have overloads for the old signatures that translate into the new signature.

Reply:
Todo added 👍 

Descore peers: https://github.com/status-im/nim-libp2p/issues/692

`ValidationResult` -> `ValidationError` issue: https://github.com/status-im/nim-libp2p/issues/691